cf-agent/package_module.c

```diff
@@ -201,219 +201,6 @@ static void ParseAndLogErrorMessage(const Rlist *data)
-        {
-            cf_pclose_full_duplex_side(io->write_fd);
-            io->write_fd = -1;
-        }
-        return 0;
-    }
-    
-    ssize_t wrt = write(io->write_fd, data, strlen(data));
-    
-    /* Make sure to close write_fd after sending all data. */
-    if (io->write_fd >= 0)
-    {
-        cf_pclose_full_duplex_side(io->write_fd);
-        io->write_fd = -1;
-    }
-    return wrt;
-}
-
-int WriteDataToPackageModule(const char *args, const char *data,
-                             const PackageModuleWrapper *wrapper)
-{
-    char *command = StringFormat("%s %s", wrapper->path, args);
-    IOData io = cf_popen_full_duplex(command, false);
-    free(command);
-    
-    if (io.write_fd == -1 || io.read_fd == -1)
-    {
-        Log(LOG_LEVEL_VERBOSE, "Error occurred while opening pipes for "
-            "communication with package module.");
-        return -1;
-    }
-
-    Log(LOG_LEVEL_DEBUG, "Opened fds %d and %d for command '%s'.",
-        io.read_fd, io.write_fd, args);
-    
-    int res = 0;
-    if (WriteScriptData(data, &io) != strlen(data))
-    {
-        Log(LOG_LEVEL_VERBOSE,
-            "Was not able to send whole data to package module.");
-        res = -1;
-    }
-    
-    /* If script returns non 0 status */
-    int close = cf_pclose_full_duplex(&io);
-    if (close != EXIT_SUCCESS)
-    {
-        Log(LOG_LEVEL_VERBOSE,
-            "Package module returned with non zero return code: %d",
-            close);
-        res = -1;
-    }
-    return res;
-}
-
-/* In some cases the response is expected to be not filled out. Some requests
-   will have response filled only in case of errors. */
-static int ReadWriteDataToPackageScript(const char *args, const char *request,
-                                        Rlist **response,
-                                        const PackageModuleWrapper *wrapper)
-{
-    assert(args && wrapper);
-    
-    char *command = StringFormat("%s %s", wrapper->path, args);
-    IOData io = cf_popen_full_duplex(command, false);
-    free(command);
-    
-    if (io.write_fd == -1 || io.read_fd == -1)
-    {
-        Log(LOG_LEVEL_INFO, "Some error occurred while communicating "
-            "package module.");
-        return -1;
-    }
-
-    Log(LOG_LEVEL_DEBUG, "Opened fds %d and %d for command '%s'.",
-        io.read_fd, io.write_fd, args);
-    
-    if (WriteScriptData(request, &io) != strlen(request))
-    {
-        Log(LOG_LEVEL_VERBOSE, "Couldn't send whole data to package module.");
-        return -1;
-    }
-    
-    /* We can have some error message here. */
-    Rlist *res = ReadDataFromPackageScript(&io);
-    
-    /* If script returns non 0 status */
-    int close = cf_pclose_full_duplex(&io);
-    if (close != EXIT_SUCCESS)
-    {
-        Log(LOG_LEVEL_VERBOSE,
-            "Package module returned with non zero return code: %d",
-            close);
-        RlistDestroy(res);
-        return -1;
-    }
-
-    *response = res;
-    return 0;
-}
-
```

Comment:
@kacf you might want to review these package module changes since it is your domain. The functionality was moved to generic functions in `pipes_unix.c`.